code/datums/outputs.dm

```diff
@@ -0,0 +1,31 @@
+GLOBAL_LIST_EMPTY(outputs_list)
+
+/datum/outputs
+	var/text = ""
+	var/list/sounds = list('sound/items/airhorn.ogg'=1) //weighted, put multiple for random selection between sounds
+	var/list/icon = list('icons/sound_icon.dmi',"circle", HUD_LAYER) //syntax: icon, icon_state, layer
+
+/datum/outputs/New()
+	GLOB.outputs_list[src.type] = src
+
+/datum/outputs/proc/send_info(mob/receiver, turf/turf_source, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
+	var/sound/S
+	if(receiver.client)
+		//Pick sound
+		if(sounds.len)
+			var/soundin = pickweight(sounds)
+			S = sound(get_sfx(soundin))
+		receiver.display_output(S, icon, text, turf_source, vol, vary, frequency, falloff, channel, pressure_affected)
```

Comment:
Create image in this proc, adjust the sound in this proc, send the final data this proc is supposed to configure
display_outputs is there is mobs need to change something from this datum, its not a default state, default is this